pysteps/xarray_helpers.py

```diff
@@ -142,25 +169,37 @@ def convert_input_to_xarray_dataset(
 
     data_vars = {
         var_name: (
-            ["y", "x"],
+            dims,
             precip,
             {
                 "units": attrs["units"],
                 "standard_name": attrs["standard_name"],
                 "long_name": attrs["long_name"],
                 "grid_mapping": "projection",
-                "transform": metadata["transform"],
-                "accutime": metadata["accutime"],
-                "threshold": metadata["threshold"],
-                "zerovalue": metadata["zerovalue"],
-                "zr_a": metadata["zr_a"],
-                "zr_b": metadata["zr_b"],
+                "xpixelsize": xpixelsize,
+                "ypixelsize": ypixelsize,
             },
         )
     }
+
+    metadata_keys = [
+        "transform",
+        "accutime",
+        "threshold",
+        "zerovalue",
+        "zr_a",
+        "zr_b",
+        "cartesian_unit",
+        "yorigin",
```

Comment:
I also don't think the `yorigin` needs to be kept as attribute any longer (at least not as attribute of the precip var), since at the end of this method we sort the dataset in the `x` and `y` dimension so the `yorigin` will always be `lower`.

Reply:
Done